klarna-on-demand/src/main/java/com/klarna/ondemand/crypto/CryptoFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.klarna.ondemand.crypto;
+
+import android.content.Context;
+import android.os.Build;
+
+public class CryptoFactory {
+
+    private static Crypto cyptoInstance;
```

Comment:
add `volatile`, as explained [here](http://stackoverflow.com/questions/7855700/why-is-volatile-used-in-this-example-of-double-checked-locking)